tests/e2e/conftest.py

```diff
@@ -0,0 +1,316 @@
+    creds_bytes = base64.b64decode(b64_creds_bytes)
+    return creds_bytes.decode("utf-8")
+
+
+@pytest.fixture(scope="module")
+def dockerhub_username(request):
+    if os.environ.get("DOCKERHUB_USERNAME") is None:
+        pytest.fail("Environment variable DOCKERHUB_USERNAME must be set")
+    return os.environ.get("DOCKERHUB_USERNAME")
+
+
+@pytest.fixture(scope="module")
+def dockerhub_password(request):
+    if os.environ.get("DOCKERHUB_PASSWORD") is None:
+        pytest.fail(
+            "Environment variable DOCKERHUB_PASSWORD must be set to your docker hub password or token"
+        )
+    return os.environ.get("DOCKERHUB_PASSWORD")
+
+
+class Helpers:
+    @staticmethod
+    async def run_cmd(unit, cmd_desc, cmd_string, fail_msg, log_results=False):
+        log.info(cmd_desc)
+        action = await unit.run(cmd_string)
+        action = await action.wait()
+        if log_results:
+            log.info(action.results)
+        code = action.results.get("Code", action.results.get("return-code"))
+        if code is None:
+            log.error(f"Failed to find the return code in {action.results}")
+            pytest.fail(f"Failed to find the return code in {action.results}")
+        if code != 0:
+            log.error(action.results)
+            pytest.fail(fail_msg)
+
+
+@pytest.fixture(scope="module")
+def helpers():
+    return Helpers
+
+
+@pytest.fixture(scope="module")
+async def microk8s_unit(
+    ops_test, helpers, proxy, metallb_ip_range, dockerhub_username, dockerhub_password
+):
+    await ops_test.model.set_config(
+        {
+            "juju-http-proxy": proxy,
+            "apt-http-proxy": proxy,
+            "snap-http-proxy": proxy,
+            "juju-https-proxy": proxy,
+            "apt-https-proxy": proxy,
+            "snap-https-proxy": proxy,
+            "apt-no-proxy": "localhost,127.0.0.1,ppa.launchpad.net,launchpad.net",
+            "juju-no-proxy": "localhost,127.0.0.1,0.0.0.0,ppa.launchpad.net,launchpad.net,10.0.8.0/24",
+            "force-vm-hardware-version": "17",
+        }
+    )
+
+    charm_config = {}
+    charm_config["containerd_env"] = "\n".join(
+        [
+            "ulimit -n 65536 || true",
+            "ulimit -l 16834 || true",
+            f"HTTP_PROXY={proxy}",
+            f"HTTPS_PROXY={proxy}",
+            "NO_PROXY=localhost,127.0.0.1,0.0.0.0,ppa.launchpad.net,launchpad.net,10.0.8.0/24,192.168.0.0/16,10.246.154.0/24,10.246.153.0/24,10.152.183.0/24",
+            f"https_proxy=h{proxy}",
+            f"http_proxy={proxy}",
+            "no_proxy=localhost,127.0.0.1,0.0.0.0,ppa.launchpad.net,launchpad.net,10.0.8.0/24,192.168.0.0/16,10.246.154.0/24,10.246.153.0/24,10.152.183.0/24",
+        ]
+    )
+    charm_config[
+        "custom_registries"
+    ] = f'[{{"host": "docker.io", "url": "registry-1.docker.io", "username": "{dockerhub_username}", "password": "{dockerhub_password}"}}]'
+    charm_config["addons"] = "dns ingress registry"
```

Comment:
The microk8s charm is being rewritten and I suspect some of these config options won't be the same. I guess there's not much to do about it right now except carry on with the understanding that these tests will need to be updated later.